source/material_model/rheology/diffusion_dislocation.cc

```diff
@@ -54,10 +55,10 @@ namespace aspect
         const double edot_ii = std::max(std::sqrt(std::max(-second_invariant(deviator(strain_rate)), 0.)),
                                         min_strain_rate);
         const double log_edot_ii = std::log(edot_ii);
-
+        double log_strain_rate_deriv;
```

Comment:
I would put that declaration right above the lambda functions that use it to minimize the scope of the variable.